File: tools/EventClients/examples/python/example_notification.py
#!/usr/bin/env python3

# This is a simple example showing how you can show a notification
# window with a custom icon inside XBMC. It could be used by mail
# monitoring apps, calendar apps, etc.

import os
from socket import *
import sys

if os.path.exists("../../lib/python"):
    # try loading modules from source directory
    sys.path.append("../../lib/python")

    from xbmcclient import *

    ICON_PATH = "../../icons/"
else:
    # fallback to system wide modules

    from kodi.xbmcclient import *
    from kodi.defs import *

def main():
    import time
    import sys

    host = "localhost"
    port = 9777
    addr = (host, port)
    sock = socket(AF_INET,SOCK_DGRAM)

    packet = PacketHELO("Email Notifier", ICON_NONE)
    packet.send(sock, addr)

    # wait for 5 seconds
    time.sleep (5)

    packet = PacketNOTIFICATION("New Mail!",             # caption
                                "RE: Check this out",    # message
                                ICON_PNG,                # optional icon type
                                ICON_PATH + "/mail.png") # icon file (local)
    packet.send(sock, addr)

    packet = PacketBYE()
    packet.send(sock, addr)

if __name__=="__main__":
    main()

File: tools/EventClients/examples/c++/example_notification.cpp
#include "../../lib/c++/xbmcclient.h"
#include <stdio.h>
#include <string.h>
#include <sys/socket.h>
#ifdef _WIN32
#include <Windows.h> // for sleep
#else
#include <unistd.h>
#endif

int main(int argc, char **argv)
{
  /* connect to localhost, port 9777 using a UDP socket
     this only needs to be done once.
     by default this is where XBMC will be listening for incoming
     connections. */

  CAddress my_addr; // Address => localhost on 9777
  int sockfd = socket(AF_INET, SOCK_DGRAM, 0);
  if (sockfd < 0)
  {
    printf("Error creating socket\n");
    return -1;
  }

  my_addr.Bind(sockfd);

  std::string sIconFile = "../../icons/mail.png";
  unsigned short usIconType = ICON_PNG;

  std::ifstream file (sIconFile, std::ios::in|std::ios::binary|std::ios::ate);
  if (!file.is_open())
  {
    sIconFile = "/usr/share/pixmaps/kodi/mail.png";
    file.open(sIconFile, std::ios::in|std::ios::binary|std::ios::ate);

    if (!file.is_open()) {
      usIconType = ICON_NONE;
    }
    else
    {
      file.close();
    }
  }
  else
  {
    file.close();
  }

  CPacketHELO HeloPackage("Email Notifier", ICON_NONE);
  HeloPackage.Send(sockfd, my_addr);

  sleep(5);

  CPacketNOTIFICATION packet("New Mail!",          // caption
                             "RE: Check this out", // message
                             usIconType,           // optional icon type
                             sIconFile.c_str());   // icon file (local)
  packet.Send(sockfd, my_addr);

  // BYE is not required since XBMC would have shut down
  CPacketBYE bye; // CPacketPing if you want to ping
  bye.Send(sockfd, my_addr);
}
